hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ReconTaskControllerImpl.java

```diff
@@ -237,8 +246,9 @@ public synchronized void stop() {
    * @param events      A batch of {@link OMUpdateEventBatch} events to fetch sequence number of last event in batch.
    * @param failedTasks Reference of the list to which we want to add the failed tasks for retry/reprocessing
    */
-  private void processTasks(Collection<NamedCallableTask<Pair<String, Boolean>>> tasks,
-                            OMUpdateEventBatch events, List<String> failedTasks) {
+  private void processTasks(
+      Collection<NamedCallableTask<ReconOmTask.TaskResult>> tasks,
+      OMUpdateEventBatch events, List<Pair<String, Map<String, Integer>>> failedTasks) {
```

Comment:
We should add index and other information to TaskResult itself, so that it can be updated instead of creaing pair, map, ...

Reply:
Handled. Pls re-review.